### ISSUE TITLE
Use unique time-full-steer for keyboard, gamepad, AI

### ISSUE THREAD

**Issue by Alayan-stk-2** [supertuxkart/stk-code#3653]
Currently, the time-full-steer, which ostensibly should be a physics characteristics rather than a control characteristics as it affects what the kart can and can't do, is not unique.

Keyboard uses the time-full-steer defined in kart_characteristics.xml
AI uses the time-full-steer (not the same !) defined in stk_config.xml
Controller with analog joystick are completely exempt of time-full-steer (see PlayerController::steer), allowing to change turn direction quicker than is otherwise possible.

It is relatively easy to patch the keyboard/joystick difference, but ensuring consistency with the AI requires more work. It is likely that the checks for full turn radius would have to be moved outside of the controllers.

**Comment by Alayan-stk-2** [comment]
The analog/keyboard case is fixed in https://github.com/supertuxkart/stk-code/commit/f3ef25d52594271f953d9db1db69a26a280309ff

The consistency with the AI remains to be done.

**Comment by qwertychouskie** [comment]
Although probably minor, I do agree that this should be handled server-side.

**Comment by Alayan-stk-2** [comment]
> The value for players and AI were designed to be separate in order to potentially help the AI
Obviously, I don't like this. The AI is good enough in SuperTux to often beat average players, and the idea of the AI "cheating" isn't pleasing. The very idea that the AI get an unfair advantage can create frustration, so I'd prefer to avoid it and be able to tell that STK's AI gets no such advantage (which is otherwise true).

However, this alone isn't high priority, and arguably the ability to use items with computer reflexes is a bigger problem for this.

> And while we don't need this (atm), I don't think there's a good reason to remove it at this stage - we can just set it to the same value.

As far as I can tell, the AI get a fixed value, while for the player it is not uniform (depending on the current steering, the rate of the steering change will be stronger or weaker, there is an interpolation array for this).

> Re networking: the controller should send the actual input event from the user, not the interpolated steering. So even is a user cheats, it should not matter since the server will overwrite the client. @Benau, do you agree?

If it is done like this, my concern is unfounded :slightly_smiling_face: 

Indeed, iirc, it's the inputs which are sent, I don't know why I thought otherwise.

**Comment by Benau** [comment]
Yes as long as everything is done in playercontroller class only then it's fine, (server has no localplayercontroller)

**Comment by Alayan-stk-2** [comment]
Moving it to the next milestone, as the gamepad handling has been made identical to the keyboard handling, which was the biggest issue.

Making the AI have similar steering to the players would be nice, but is not essential.

**Comment by Alayan-stk-2** [comment]
Done in the Evolution branch with https://github.com/supertuxkart/stk-code/commit/99ef9d00761965a45a118ea85ae7339e4309c241
